src/hooks/useVotesRevealedEvents.ts

```diff
@@ -1,29 +1,17 @@
-import { useContext } from "react";
 import { useQuery } from "react-query";
 import { ethers } from "ethers";
 import {
   queryVotesRevealedEvents,
   VoteRevealed,
 } from "common/web3/get/queryVotesRevealedEvents";
-import { ErrorContext } from "common/context/ErrorContext";
 
 export default function useVotesRevealedEvents(
   contract: ethers.Contract | null,
   address: string | null
 ) {
-  const { addError } = useContext(ErrorContext);
-
-  const { data, error, isFetching, refetch } = useQuery<
-    VoteRevealed[] | undefined | void
-  >(
-    "votesRevealedEvents",
-    () => {
-      return queryVotesRevealedEvents(contract, address)
-        .then((res) => res)
-        .catch((err) => addError(err));
-    },
-    { enabled: contract !== null }
+  return useQuery<VoteRevealed[] | undefined | void>(
+    ["votesRevealedEvents", address],
+    () => queryVotesRevealedEvents(contract, address),
+    { enabled: !!contract && !!address }
```

Comment:
this was just refactored to be shorter and remove the error notification

Reply:
Does contract also need to be part of the key?

Reply:
I would avoid using contract in the key since its not a primitive value and theres a chance the contract instance could change without changing the underlying contract address or user address, triggering an unecessary refetch.   I would need to be really sure that `contract` is not changing due to some weird useEffect where its initialized.